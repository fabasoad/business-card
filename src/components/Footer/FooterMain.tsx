import * as React from 'react'
import { useTranslation } from 'react-i18next'

import FooterContacts from './FooterContacts'
import FooterExtra from './FooterExtra'
import FooterInfo from './FooterInfo'
import FooterSocial from './FooterSocial'

export default function FooterMain() {
  const { t } = useTranslation()
  return (
    <div className="footer dark-component">
      <div id="contact" className="container text-center">
        <div className="section-title center">
          <h2>{t('business-card-contact-title')}</h2>
          <hr />
        </div>
        <FooterContacts />
        <FooterSocial />
        <FooterInfo />
        <FooterExtra />
      </div>
    </div>
  )
}
