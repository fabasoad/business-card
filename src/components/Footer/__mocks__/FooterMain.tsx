import * as React from 'react'

export default function FooterMain() {
  return (
    <div data-testid="FooterMain"></div>
  )
}
