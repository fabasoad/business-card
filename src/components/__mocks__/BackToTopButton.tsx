import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
