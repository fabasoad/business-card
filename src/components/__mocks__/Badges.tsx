import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
