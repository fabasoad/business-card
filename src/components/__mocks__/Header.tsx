import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
