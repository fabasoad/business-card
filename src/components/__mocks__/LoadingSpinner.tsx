import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
