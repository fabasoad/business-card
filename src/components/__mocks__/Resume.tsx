import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
