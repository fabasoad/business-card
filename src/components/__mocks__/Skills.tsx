import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
