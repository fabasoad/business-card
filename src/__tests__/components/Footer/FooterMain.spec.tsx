/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import FooterMain from '../../../components/Footer/FooterMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render FooterMain correctly', () => {
  const wrapper: ShallowWrapper = shallow(<FooterMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith('business-card-contact-title')
})
