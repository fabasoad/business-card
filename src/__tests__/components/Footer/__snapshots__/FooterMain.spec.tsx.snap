// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FooterMain correctly 1`] = `
<div
  className="footer dark-component"
>
  <div
    className="container text-center"
    id="contact"
  >
    <div
      className="section-title center"
    >
      <h2>
        business-card-contact-title
      </h2>
      <hr />
    </div>
    <FooterContacts />
    <FooterSocial />
    <FooterInfo />
    <FooterExtra />
  </div>
</div>
`;
